client/dashboard/domains/name-servers/index.tsx

```diff
@@ -0,0 +1,46 @@
+import { useMutation, useQuery } from '@tanstack/react-query';
+import { Card, CardBody } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useAppContext } from '../../app/context';
+import {
+	domainNameServersQuery,
+	domainNameServersMutation,
+} from '../../app/queries/domain-name-servers';
+import { domainRoute } from '../../app/router';
+import { PageHeader } from '../../components/page-header';
+import PageLayout from '../../components/page-layout';
+import NameServersForm from './form';
+import './styles.scss';
+
+export default function NameServers() {
+	const { instanceType } = useAppContext();
+	const { domainName } = domainRoute.useParams();
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const { data: nameServers, error: queryError } = useQuery( domainNameServersQuery( domainName ) );
```

Comment:
Is this domain name servers query something we should fetch in the router? That way we can use suspense here

```suggestion
	const { data: nameServers, error: queryError } = useSuspenseQuery( domainNameServersQuery( domainName ) );
```

Reply:
I would not fetch name servers data in the router with Suspense because we want fine-grained control over error handling in the UI.

Using Suspense with useSuspenseQuery would force us to handle errors at a route level through error boundaries, which would be a worse user experience than our current approach of showing inline errors following design.

Reply:
Interesting, yeah the design has this new pattern for showing error 

QOBjujZah0UlGDDdLKZhzi-fi-10344_149770


<img width="574" height="250" alt="Image" src="https://github.com/user-attachments/assets/53b77404-edfb-49f0-a5fb-5e107bf4fa4a" />

It's a new thing that we have not used in existing code; we can rethink later on how we should handle this case 🤔